cachito/workers/pkg_managers/rubygems.py

```diff
@@ -376,6 +386,26 @@ def resolve_rubygems(package_root, request):
     }
 
 
+def unpack_and_checkout_proper_branch(dep):
+    """
+    Unpack the archive with the downloaded dependency and checkout the proper branch.
+
+     so that local git repo dependency replacement works.
+
+    Only the unpacked directory is kept, the archive is deleted.
+    To get more info on local Git repos, see:
+    https://bundler.io/man/bundle-config.1.html#LOCAL-GIT-REPOS
+    :param dep: RubyGems GIT dependency
+    """
+    #
+    extracted_path = str(dep["path"]).removesuffix(".tar.gz")
+    shutil.unpack_archive(dep["path"], extracted_path)
+    os.remove(dep["path"])
+    dep["path"] = extracted_path
+
+    subprocess.run(f"cd {extracted_path}/app && git switch {dep['branch']}", check=True)
```

Comment:
Can I execute this command or do I need to use something like GitPython (Bandit is complaining)?

Reply:
Is the checkout necessary? The archive cloned by Cachito should be checked out at the exact ref specified when creating the `Git` object

https://github.com/containerbuildsystem/cachito/blob/35fb8db4a8146422a8772f4fb5b261c2cbaef2b2/cachito/workers/pkg_managers/rubygems.py#L317

Reply:
I will double check this and let you know, but I tested it manually without this at first, and I was always receiving errors that there's no branch checked out in the local git repo. But looking at it, yes it's true that the right commit will be checked out based on commit hash, but it will be just a detached head and not a branch (we're not passing any information about the exact branch), which confuses Bundler.

Reply:
You were right, [there's an option disabling branch checks](https://bundler.io/man/bundle-config.1.html#CREDENTIALS-FOR-GEM-SOURCES): `BUNDLE_DISABLE_LOCAL_BRANCH_CHECK: "true"`

Therefore, no checkout will be necessary.